lib/ansible/modules/cloud/amazon/aws_codedeploy_application.py

```diff
@@ -0,0 +1,438 @@
+    new_deployment_group = module.params.get('new_deployment_group')
+    auto_scaling_groups = module.params.get('auto_scaling_groups')
+    trigger_configs = module.params.get('trigger_configs')
+    deployment_config = module.params.get('deployment_config')
+    ec2_tag_filters = module.params.get('ec2_tag_filters')
+    on_premise_filters = module.params.get('on_premise_filters')
+    alarm_configuration = module.params.get('alarm_configuration')
+    auto_rollback_config = module.params.get('auto_rollback_config')
+    bg_deployment_config = module.params.get('bg_deployment_config')
+    load_balance_info = module.params.get('load_balance_info')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+    connection = boto3_conn(module, conn_type='client', resource='codedeploy', region=region, endpoint=ec2_url, **aws_connect_params)
+
+    changed = False
+    if state == 'present':
+        # Check if the application already exists, if it does, update it.
+        has_application = get_codedeploy_application(connection, name, module)
+        # Check if the deployment group already exists, if it does, update it.
+        has_deployment_group = get_codedeploy_application_deployment_group(connection, name, deployment_group, module)
+        if not has_application:
+            cd_properties = create_codedeploy_application(
+                connection,
+                name,
+                deployment_group,
+                new_deployment_group,
+                deployment_config,
+                ec2_tag_filters,
+                on_premise_filters,
+                auto_scaling_groups,
+                service_role_arn,
+                trigger_configs,
+                alarm_configuration,
+                auto_rollback_config,
+                deployment_style,
+                bg_deployment_config,
+                load_balance_info,
+                module)
+        else:
+            # Update the application name if a new name was given.
+            if new_application_name:
+                cd_properties = update_codedeploy_application(connection, name, new_application_name, module)
+        if has_deployment_group:
+            cd_properties = update_codedeploy_deployment_group(
+                connection,
+                name,
+                deployment_group,
+                new_deployment_group,
+                deployment_config,
+                ec2_tag_filters,
+                on_premise_filters,
+                auto_scaling_groups,
+                service_role_arn,
+                trigger_configs,
+                alarm_configuration,
+                auto_rollback_config,
+                deployment_style,
+                bg_deployment_config,
+                load_balance_info,
+                module)
+        else:
+            cd_properties = create_codedeploy_deployment_group(
```

Comment:
These should probably be keyword parameters - a massive length of ordered parameters is an easy source of bugs.